### PR TITLE
[CALCITE-1683] Druid Specific Filter Transpose Rules

### PR DESCRIPTION
**Pull request by nishantmonu51**

### CODE REVIEW

**Comment by nishantmonu51** [comment]
@julianhyde @jcamachor: 
One test which is still failing after this patch is DruidAdapterTest.testfilterTimestamp - 
The plan generated with patch is below - Which generates a timeseries instead of a select query on druid which I think is good, but the intervals are not being pushed correctly to druid query with or without the patch, so it seems something is broken with https://issues.apache.org/jira/browse/CALCITE-1334

Plan after patch - 
```
PLAN=EnumerableInterpreter
  BindableAggregate(group=[{}], C=[$SUM0($1)])
    BindableFilter(condition=[AND(=(EXTRACT_DATE(FLAG(YEAR), /INT(Reinterpret($0), 86400000)), 1997), OR(=(EXTRACT_DATE(FLAG(MONTH), /INT(Reinterpret($0), 86400000)), 4), =(EXTRACT_DATE(FLAG(MONTH), /INT(Reinterpret($0), 86400000)), 6)))])
      DruidQuery(table=[[foodmart, foodmart]], intervals=[[1900-01-09T00:00:00.000/2992-01-10T00:00:00.000]], groups=[{0}], aggs=[[COUNT()]])
```

File: druid/src/test/java/org/apache/calcite/test/DruidAdapterIT.java [hunk 1]
```diff
@@ -1207,10 +1207,18 @@ public Void apply(ResultSet resultSet) {
         + "EnumerableInterpreter\n"
         + "  BindableSort(sort0=[$1], dir0=[DESC], fetch=[2])\n"
         + "    BindableProject(state_province=[$0], CDC=[FLOOR($1)])\n"
-        + "      BindableAggregate(group=[{1}], agg#0=[COUNT($0)])\n"
-        + "        DruidQuery(table=[[foodmart, foodmart]], intervals=[[1900-01-09T00:00:00.000/2992-01-10T00:00:00.000]], groups=[{29, 30}], aggs=[[]])";
+        + "      DruidQuery(table=[[foodmart, foodmart]], intervals=[[1900-01-09T00:00:00.000/2992-01-10T00:00:00.000]], groups=[{30}], aggs=[[COUNT(DISTINCT $29)]])\n";
```
Comment (nishantmonu51) [inline druid/src/test/java/org/apache/calcite/test/DruidAdapterIT.java]: test comment says that it tests that distinct count is pushed down to druid, but actually it was not pushed down to druid.

**Comment by julianhyde** [comment]
I thought that test was fixed in master.

**Comment by jcamachor** [comment]
@nishantmonu51 , changes LGTM.

**Pull request closed without merging**